internal/multistorage/cache/status_cache.go

```diff
@@ -0,0 +1,160 @@
+package cache
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/wal-g/wal-g/pkg/storages/storage"
+)
+
+type StatusCache struct {
+	storages []NamedFolder
+	ttl      time.Duration
+}
+
+func NewStatusCache(primary storage.Folder, failover map[string]storage.Folder, ttl time.Duration) *StatusCache {
+	return &StatusCache{
+		storages: nameFolders(primary, failover),
+		ttl:      ttl,
+	}
+}
+
+func (c *StatusCache) AllAliveStorages() ([]NamedFolder, error) {
+	memCacheMu.Lock()
+	defer memCacheMu.Unlock()
+
+	if memCache.isRelevant(c.ttl, c.storages...) {
+		return memCache.getAllAlive(c.storages), nil
+	}
+
+	oldFile, err := readFile()
+	if err != nil {
+		return nil, fmt.Errorf("read cache file: %w", err)
```

Comment:
I guess that reading the cache file should not produce an error of the entire operation. Just ignore and overwrite it?